book/src/simple_gun_weapon.md

```diff
@@ -0,0 +1,212 @@
+            SNIPER_COLLIDER_WIDTH,
+        );
+
+        let gun_fx_sprite = GunlikeAnimation::new(
+            AnimatedSprite::new(
+                92,
+                32,
+                &[Animation {
+                    name: "shoot".to_string(),
+                    row: 2,
+                    frames: 3,
+                    fps: 15,
+                }],
+                false,
+            ),
+            resources.items_textures["sniper/gun"], // Change this to `your_weapon/gun`
+            SNIPER_COLLIDER_WIDTH,
+        );
+
+        scene::add_node(Gun {
+            gun_sprite,
+            gun_fx_sprite,
+            gun_fx: false,
+            body: PhysicsBody::new(
+                &mut resources.collision_world,
+                pos,
+                0.0,
+                vec2(SNIPER_COLLIDER_WIDTH, SNIPER_COLLIDER_HEIGHT),
+            ),
+            throwable: ThrowableItem::default(),
+            bullets: SNIPER_BULLETS,
+            max_bullets: SNIPER_BULLETS,
+            bullet_speed: SNIPER_BULLET_SPEED,
+            collider_width: SNIPER_COLLIDER_WIDTH,
+            collider_height: SNIPER_COLLIDER_HEIGHT,
+            recoil: SNIPER_RECOIL,
+        })
+        .untyped()
+    }
+}
+```
+
+### Items Array
+
+Open `src/items.rs` and add find the `ITEMS` array. Add an entry to this array for your weapon using your new spawn function. I copied the `Item` entry for the musket weapon and changed the `tiled_name` and `constructor` fields to `sniper` and `gun::Gun::spawn_sniper` respectively. This names my item "sniper" and will call the spawn function we programmed above to spawn the sniper weapon. The following is the entry for the sniper weapon:
+
+```rust
+Item {
+    tiled_name: "sniper",
+    constructor: gun::Gun::spawn_sniper,
+    tiled_offset: (-35., -25.),
+    textures: &[("gun", "assets/Whale/Gun(92x32).png")], // Temporarily using the existing gun texture
+    sounds: &[],
+    fxses: &[],
+    network_ready: true,
+},
+```
+
+If you like, you can skip to the testing section to test your new weapon, but if you haven't added a new texture or modified an existing texture, you will not be able to visually recognize your item before picking it up. For this reason, I recommend you continue to the [Texture](#texture) section next.
+
+## Texture
+
+It is important to make sure that your weapon is able to be visually distinguished between the other weapons in the game. When I added the sniper rifle to the `ITEMS` array, I copied all of the data from existing musket weapon, changing only `tiled_name` and `constructor`. To give my sniper weapon a new texture, I will also need to change the "gun" texture in the `textures` field. Currently, the "gun" texture is set to `assets/Whale/Gun(92x32).png`. I'll open this file with my pixel editor of choice, [Aseprite](https://www.aseprite.org/) ([GIMP](https://www.gimp.org/) would also work fine).
```

Comment:
Could use an explanation for what `tiled_name` does (must match name in the map)

Reply:
'tiled_name' should probably be changed to 'item_id' as we are moving away from Tiled. I haven't added items to the map editor yet, but when I do, I will probably have to change this, as well as create a way to define these in a data file, for reference by the editor and the scene builder